From 10d55f09b5205a5bbc96674f0409093c2d118db7 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:03 -0700
Subject: [PATCH 1627/1851] v4l: xilinx: sdirxss: Fix warnings for UNUSED_VALUE

commit 88346b30e50286da0b39e80b92989026d21281cd from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNUSED_VALUE type.

The actual warnings are:
For xsdirx_setedherrcnttrigger():
Event returned_value: 	Assigning value from "xsdirxss_read(core, 80U)"
to "val" here, but that stored value is overwritten before it can be
used.
Event value_overwrite: 	Overwriting previous write to "val" with value
from "enable & 0xffffU".

For xsdirxss_set_gtclk():
Event returned_value: 	Assigning value from "clk_get_rate(gtclk)" to
"clkrate" here, but that stored value is overwritten before it can be
used.
Event value_overwrite: 	Overwriting previous write to "clkrate" with
value "148500000UL".
Event value_overwrite: 	Overwriting previous write to "clkrate" with
value "148351648UL".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 8ac72785accf..85a1df083078 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -854,7 +854,7 @@ static void xsdirx_framer(struct xsdirxss_core *core, bool flag)
 
 static void xsdirx_setedherrcnttrigger(struct xsdirxss_core *core, u32 enable)
 {
-	u32 val = xsdirxss_read(core, XSDIRX_EDH_ERRCNT_EN_REG);
+	u32 val;
 
 	val = enable & XSDIRX_EDH_ALLERR_MASK;
 
@@ -1034,7 +1034,6 @@ static void xsdirxss_set_gtclk(struct xsdirxss_state *state)
 
 	/* get sdi_rx_clk */
 	gtclk = core->clks[1].clk;
-	clkrate = clk_get_rate(gtclk);
 	is_frac = state->frame_interval.numerator == 1001 ? 1 : 0;
 
 	/* calcualte clkrate */
-- 
2.31.1

